topo/topo.go

```diff
@@ -461,6 +481,7 @@ type TopologyParams struct {
 	TopoNewOptions []Option // the options used in the TopoNewFunc
 	Timeout        time.Duration
 	DryRun         bool
+	fakeTopo       *fakeTopology // a fake topology manager for testing only
```

Comment:
something about this feels strange to me, but it doesn't exactly make sense as a topoNewOption either